backend/core/src/cron/cron.service.ts

```diff
@@ -1,10 +1,45 @@
 import { Injectable } from "@nestjs/common"
 import { Cron, CronExpression } from "@nestjs/schedule"
+import { UserListQueryParams } from "../auth/dto/user-list-query-params"
+import { UserService } from "../auth/services/user.service"
+import { AuthContext } from "../auth/types/auth-context"
+import { Compare } from "../shared/dto/filter.dto"
+import { EmailService } from "../shared/email/email.service"
+import { ListingsService } from "../listings/listings.service"
 
 @Injectable()
 export class CronService {
-  @Cron(CronExpression.EVERY_WEEK)
-  handleCron() {
-    // TODO: add a cron task here.
+  constructor(
+    private readonly emailService: EmailService,
+    private readonly listingsService: ListingsService,
+    private readonly userService: UserService
+  ) {}
+
+  @Cron(CronExpression.EVERY_1ST_DAY_OF_MONTH_AT_NOON)
+  async handleCron() {
+    if (!process.env.SEND_NOTIFICATIONS_FOR_UPDATE_LISTINGS_REMINDER) {
+      return
+    }
+
+    const userQueryParams: UserListQueryParams = {
+      filter: [
+        {
+          $comparison: Compare["EQUALS"],
+          isPartner: true,
+        },
+      ],
+    }
+    const users = await this.userService.list(userQueryParams, new AuthContext())
+    const allListings = await this.listingsService.list({})
+
+    // For each listing, check whether the listed leasing agents are in the list of partner users.
+    // If the leasing agent is a partner and has their email notifications turned on, send the reminder email.
+    for (const listing of allListings.items) {
+      for (const leasingAgent of listing.leasingAgents) {
+        if (users.items.includes(leasingAgent) && leasingAgent.preferences.sendEmailNotifications) {
+          await this.emailService.updateListingReminder(listing, leasingAgent)
```

Comment:
Does the sendgrid API support sending a batch of emails? It seems like we could get to pretty long run times pretty quickly if we need to wait for each individual email to be sent.

Reply:
Yeah it does support batch, I can implement it this way if we can assume that all partners will be english speakers. If we want multiple languages the issue here is that we load translations given a single user at the beginning of each send email function - that means if we pass in like 20 users we would have to sort them by language first probably and then send batches for each language

Reply:
If it's not too complicated, I would do batch for now and add a note about translations and let Exygy decide what to do. If it's too much (more than a day of work) then just add a note here about looking into batch sending

Reply:
Yeah I can do the batch pretty quickly if its just English! I'll add a note about the translations